aspnetcore/blazor/security/webassembly/standalone-with-identity.md

```diff
@@ -208,11 +214,30 @@ Call <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProv
 > * <xref:blazor/call-web-api>
 > * <xref:blazor/security/webassembly/additional-scenarios>
 
+## Test user seeding demonstration
+
+The `SeedData` class ([`SeedData.cs`](https://github.com/dotnet/blazor-samples/blob/main/8.0/BlazorWebAssemblyStandaloneWithIdentity/Backend/SeedData.cs)) demonstrates how to create test users for development. The test user, named Leela, signs into the app with the email address `leela@contoso.com`. The user's password is set to `Passw0rd!`. Leela is given `Administrator` and `Manager` roles for authorization, which enables the user to access the manager page at `/private-manager-page` but not the editor page at `/private-editor-page`.
+
+> [!WARNING]
+> Never allow test user code to run in a production environment. `SeedData.InitializeAsync` is only called in the `Development` environment in the `Program` file:
+>
+> ```csharp
+> if (builder.Environment.IsDevelopment())
+> {
+>     await using var scope = app.Services.CreateAsyncScope();
+>     await SeedData.InitializeAsync(scope.ServiceProvider);
+> }
+> ```
+
 ## Roles
 
-For security reasons, role claims aren't sent back from the `manage/info` endpoint to create `UserInfo.Claims` for users of the `BlazorWasmAuth` app.
+For security reasons, role claims aren't sent back from the `manage/info` endpoint to create user claims for users of the `BlazorWasmAuth` app. Role claims are managed independently via a separate request in the `GetAuthenticationStateAsync` method of the [`CookieAuthenticationStateProvider` class (`Identity/CookieAuthenticationStateProvider.cs`)](https://github.com/dotnet/blazor-samples/blob/main/8.0/BlazorWebAssemblyStandaloneWithIdentity/BlazorWasmAuth/Identity/CookieAuthenticationStateProvider.cs) after the user is authenticated in the `Backend` project. 
```

Comment:
What are these security reasons? Could you be very specific here?

Reply:
I think it's CRIME/BREACH types of attacks. I'll see if I can look that up quickly and squeeze it on here.

Reply:
Chris said it's this ...

https://github.com/dotnet/aspnetcore/pull/51177

... and that's not what I thought I was going to find 😆. Let me look that over and see ... I might be able to cross-link it directly.

Reply:
Yeah ... I don't have a problem just cross-linking the issue here. Let's let readers see what happened and why they did what they did. I'll also leave a tracking note to myself for 9.0 because I bet that they have a plan to fix this. If they fix it, we'll update for role claims ***BUT*** leave an example like what we're doing here for role claims that can be used for other claim types. We'll deal with that when we circle back. For now, I'll cross-link the issue so devs can read all about it. I'll also change the "security reasons" language to something more appropriate.